syntax = "proto3";
 
message RegionSpeedsRequest {
    string region_name = 1;
    bool mean = 2;
}

message RegionSpeeds {
    string country_name = 1;
    string country_code = 2;
    string region_name = 3;
    string region_code = 4;
    string date_request = 5;
    string isp_name = 6;
    string download_speed = 7;
    string upload_speed = 8; 
}

message RegionSpeedsResponse {
    repeated RegionSpeeds regionSpeeds = 1;
}

service RegionSpeedsService {
    rpc RegionSpeedsInfo (RegionSpeedsRequest) returns (RegionSpeedsResponse);
}